Dockerfile.konflux

```diff
@@ -0,0 +1,111 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+ARG GOLANG_VERSION=1.21
+ARG BUILD_BASE=develop
+
+FROM --platform=$BUILDPLATFORM registry.redhat.io/ubi8/go-toolset@sha256:4ec05fd5b355106cc0d990021a05b71bbfb9231e4f5bdc0c5316515edf6a1c96 AS build
+
+FROM --platform=$BUILDPLATFORM $BUILD_BASE AS build
+
+LABEL image="build"
+
+USER root
+
+# needed for konflux as the previous stage is not used
+WORKDIR /opt/app
+COPY go.mod go.sum ./
+# Download dependencies before copying the source so they will be cached
+RUN go mod download
+
+# Copy the source
+COPY . ./
+
+# https://docs.docker.com/engine/reference/builder/#automatic-platform-args-in-the-global-scope
+# don't provide "default" values (e.g. 'ARG TARGETARCH=amd64') for non-buildx environments,
+# see https://github.com/docker/buildx/issues/510
+ARG TARGETOS
+ARG TARGETARCH
+
+# Build the binaries using native go compiler from BUILDPLATFORM but compiled output for TARGETPLATFORM
+# https://www.docker.com/blog/faster-multi-platform-builds-dockerfile-cross-compilation-guide/
+RUN --mount=type=cache,target=/root/.cache/go-build \
+    --mount=type=cache,target=/go/pkg \
+    export GOOS=${TARGETOS:-linux} && \
+    export GOARCH=${TARGETARCH:-amd64} && \
+    go build -o puller model-serving-puller/main.go && \
+    go build -o triton-adapter model-mesh-triton-adapter/main.go && \
+    go build -o mlserver-adapter model-mesh-mlserver-adapter/main.go && \
+    go build -o ovms-adapter model-mesh-ovms-adapter/main.go && \
+    go build -o torchserve-adapter model-mesh-torchserve-adapter/main.go
+
+
+###############################################################################
+# Stage 3: Copy build assets to create the smallest final runtime image
+###############################################################################
+FROM registry.access.redhat.com/ubi8/ubi-minimal:latest as runtime
+
+ARG USER=2000
+
+USER root
+
+# install python to convert keras to tf
+# NOTE: tensorflow not supported on PowerPC (ppc64le) or System Z (s390x) https://github.com/tensorflow/tensorflow/issues/46181
+RUN --mount=type=cache,target=/root/.cache/microdnf:rw \
+    microdnf install --setopt=cachedir=/root/.cache/microdnf --setopt=ubi-8-appstream-rpms.module_hotfixes=1 \
+       gcc \
+       gcc-c++ \
+       python38-devel \
+       python38 \
+    && ln -sf /usr/bin/python3 /usr/bin/python \
+    && ln -sf /usr/bin/pip3 /usr/bin/pip \
+    && true
+
+# need to upgrade pip and install wheel before installing grpcio, before installing tensorflow on aarch64
+# use caching to speed up multi-platform builds
+COPY requirements.txt requirements.txt
+ENV PIP_CACHE_DIR=/root/.cache/pip
+RUN --mount=type=cache,target=/root/.cache/pip \
+    pip install -r requirements.txt
```

Comment:
where is the `requirements.txt`?
will this be added later?

Reply:
I see the requirements.txt have been already added by you when we did the PoC work